tests/transforms/test_split_non_commuting.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Tests for the transform ``qml.transform.split_non_commuting()`` """
+import pytest
+import numpy as np
+import pennylane as qml
+import pennylane.numpy as pnp
+
+from pennylane.transforms import split_non_commuting
+
+
+
+
+# Unit tests for split_non_commuting
+
+
+def test_commuting_group_no_split():
+    """Testing that commuting groups are not split"""
+    with qml.tape.QuantumTape() as tape:
+        qml.PauliZ(0)
+        qml.Hadamard(0)
+        qml.CNOT((0, 1))
+        qml.expval(qml.PauliZ(0))
+        qml.expval(qml.PauliZ(0))
+        qml.expval(qml.PauliX(1))
+        qml.expval(qml.PauliZ(2))
+        qml.expval(qml.PauliZ(0) @ qml.PauliZ(3))
+
+    split, fn = split_non_commuting(tape)
+
+    assert split == [tape]
+    assert all([isinstance(t, qml.tape.QuantumTape) for t in split])
+    assert (
+        fn([0.5]) == 0.5
+    )  # is there a better way to assert that fn is just taking the first element of what ever it is input?
```

Comment:
yes! I think you might be able to use [`mocker`](https://github.com/pytest-dev/pytest-mock) to 'spy' on something that you know shouldn't be called.

e.g., 
```python
def test_commuting_group_no_split(mocker):
    ...
    spy = mocker.spy(qml.math, "concatenate")
    split, fn = split_non_commuting(tape)
    fn([0.5]) == 0.5  # execute the function
    spy.assert_not_called()
```

(note: my syntax/usage above might not be entirely correct, just typing this out from memory)